File: Services/services/protocols/webapi.js
var webapiClient = require('./../library/webapi_client');
var models       = require('./models');
exports = {};
// signup
// サインアップAPI
exports.signup = function(callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    return client.post("/signup", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
// signin
// サインインAPI
exports.signin = function(signinToken, callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    data["signinToken"] = signinToken; // サインイントークン
    return client.post("/signin", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
// rename
// 名前変更API
exports.rename = function(name, callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    data["name"] = name; // 変更する名前
    return client.post("/rename", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
// matching
// マッチングをリクエスト
exports.matching = function(callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    return client.post("/matching", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
// player
// プレイヤー情報の取得
exports.player = function(callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    return client.post("/player", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
// test
// ユニットテスト用
exports.test = function(reqValue, callback, queries = null, forms = null, headers = null) {
    var client = webapiClient.getClient();
    var data = {};
    data["reqValue"] = reqValue; // リクエストの値
    return client.post("/test", data, function(err, responseData) {
        if (err) {
            if (callback != null) {
                callback(err, null);
            }
            return;
        }
        if (callback != null) {
            callback(err, responseData);
        }
    }, queries, forms, headers);
}
module.exports = exports;

File: Assets/Game/Scripts/Services/Protocols/WebAPI.cs
using System;
using System.Collections;
using System.Collections.Generic;
using Services.Protocols.Models;
using Services.Protocols.Consts;
using UnityEngine;
namespace Services.Protocols {
    public class WebAPI {
        [Serializable]
        public struct SignupRequest {
        }

        [Serializable]
        public struct SignupResponse {
        }

        // Signup
        // サインアップAPI
        public static WebAPIClient.Request Signup(Action<string,SignupResponse> callback, string[] queries = null, string[] forms = null, string[] headers = null) {
            var client = WebAPIClient.GetClient();
            var data = new SignupRequest();
            return client.Post<SignupRequest,SignupResponse>("/signup", data, callback, queries, forms, headers);
        }
        [Serializable]
        public struct SigninRequest {
            public string signinToken; // サインイントークン
        }

        [Serializable]
        public struct SigninResponse {
        }

        // Signin
        // サインインAPI
        public static WebAPIClient.Request Signin(string signinToken, Action<string,SigninResponse> callback, string[] queries = null, string[] forms = null, string[] headers = null) {
            var client = WebAPIClient.GetClient();
            var data = new SigninRequest();
            data.signinToken = signinToken; // サインイントークン
            return client.Post<SigninRequest,SigninResponse>("/signin", data, callback, queries, forms, headers);
        }
        [Serializable]
        public struct MatchingRequest {
        }

        [Serializable]
        public struct MatchingResponse {
        }

        // Matching
        // マッチングをリクエスト
        public static WebAPIClient.Request Matching(Action<string,MatchingResponse> callback, string[] queries = null, string[] forms = null, string[] headers = null) {
            var client = WebAPIClient.GetClient();
            var data = new MatchingRequest();
            return client.Post<MatchingRequest,MatchingResponse>("/matching", data, callback, queries, forms, headers);
        }
        [Serializable]
        public struct PlayerRequest {
        }

        [Serializable]
        public struct PlayerResponse {
        }

        // Player
        // プレイヤー情報の取得
        public static WebAPIClient.Request Player(Action<string,PlayerResponse> callback, string[] queries = null, string[] forms = null, string[] headers = null) {
            var client = WebAPIClient.GetClient();
            var data = new PlayerRequest();
            return client.Post<PlayerRequest,PlayerResponse>("/player", data, callback, queries, forms, headers);
        }
        [Serializable]
        public struct TestRequest {
            public int reqValue; // リクエストの値
        }

        [Serializable]
        public struct TestResponse {
            public int resValue; // レスポンスの値
        }

        // Test
        // ユニットテスト用
        public static WebAPIClient.Request Test(int reqValue, Action<string,TestResponse> callback, string[] queries = null, string[] forms = null, string[] headers = null) {
            var client = WebAPIClient.GetClient();
            var data = new TestRequest();
            data.reqValue = reqValue; // リクエストの値
            return client.Post<TestRequest,TestResponse>("/test", data, callback, queries, forms, headers);
        }
    }
}
